### PR TITLE
do the phedex file injection in chunks

### PR DESCRIPTION
**Pull request by hufnagel**
Tier0 had a huge backlog of 150k uninjected files, which the PhEDExInjector tried to inject in a single data service call. Obviously that didn't succeed. Broke this up into chunks. Limits are 20 datasets, 50 blocks or 5000 files. Whenever we go over this (in units of all files for a dataset), we inject.

### CODE REVIEW

**Comment by hufnagel** [comment]
This is deployed in the production Tier0.

**Comment by amaltaro** [comment]
Without really looking at the code, about "Whenever we go over this (in units of all files for a dataset), we inject.", does it mean files will be injected ONLY when we go over these limits?

**Comment by hufnagel** [comment]
We also inject at the end of the polling cycle if we are still under the limits :-)

**Comment by hufnagel** [comment]
retest this please

**Comment by amaltaro** [comment]
I was wrong, it does inject data even if not over the limits. It will always - at least - inject one dataset for one site.

File: src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py [hunk 1]
```diff
@@ -206,46 +205,84 @@ def injectFiles(self):
...
-                # If we get an error here, assume that it's temporary (it usually is)
-                # log it, and ignore it in the algorithm() loop
-                msg =  "Encountered error while attempting to inject blocks to PhEDEx.\n"
-                msg += str(ex)
-                logging.error(msg)
-                logging.debug("Traceback: %s" % str(traceback.format_exc()))
-                
-                raise PhEDExInjectorPassableError(msg)
-            logging.info("Injection result: %s" % injectRes)
+            maxDatatset = 20
```
Comment (amaltaro) [inline src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py]: typo

Comment (hufnagel) [inline src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py]: It's not a typo if you get it wrong everywhere :-). But yes, wasn't supposed to look like this, will fix.

**Comment by hufnagel** [comment]
Seems these unit tests are unstable...

**Comment by ericvaandering** [comment]
I've never seen that one pop up before. But now there is a list of unstable unit tests in WMCore/test/etc You can make a pull request against that.

**Comment by ticoann** [comment]
retest this please

**Comment by ericvaandering** [comment]
Needs to be rebased.

**Comment by hufnagel** [comment]
retest this please

**Comment by hufnagel** [comment]
@amaltaro , please include this in your testing.

**Comment by hufnagel** [comment]
From the list of pylint complains the only thing I am considering dealing with is to remove the unneeded myThread. Everything else is related to our general coding style. What do you think, is the myThread worth changing ?

**Comment by ericvaandering** [comment]
The string arguments to logging may be worth changing too. Of course, as we write new code or really work old code, we should pay attention to the init warnings. But I agree it's not worth re-working.

**Comment by hufnagel** [comment]
Ok, done. Might as well get used to using the new style...

**Comment by hufnagel** [comment]
retest this please

**Comment by hufnagel** [comment]
retest this please

**Comment by hufnagel** [comment]
https://cmssdt.cern.ch/jenkins/job/DMWM-PR-test/816/testReport/junit/WMCore_t.WorkQueue_t.WorkQueue_t/WorkQueueTest/testProductionMultiQueue/

seems to be broken in general. I consider this PR complete and working.

**Comment by hufnagel** [comment]
@ticoann , are we waiting on anything to merge this ?

**Comment by hufnagel** [comment]
Ok, fixed the merge conflict, lets wait for the next jenkins test and then if it looks good please merge it.

**Pull request merged**